### PR TITLE
Add github actions replacement for failing travis closes #132

### LINKED ISSUE #132: Travis is failing to build Money
`java.net.ConnectException: Failed to connect to repo.typesafe.com/34.202.110.138:80`

### PR DESCRIPTION
**Pull request by ptravers**
Refactors our CI to run in Github Actions

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run tests
+      run: sbt clean coverage test coverageReport
+    - name: Coverage Aggregate
+      run: sbt coverageReport
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v1
+      with:
+        fail_ci_if_error: true
```
Comment (pauljamescleary) [inline .github/workflows/ci.yml:23]: This doesn't cache.

Something like this (pulled from vinyldns) would include caching and easier to compile against multiple jdks...

```
# Much copied from sbt-github-actions, modified to support running e2e tests
name: Continuous Integration

on:
  pull_request:
    branches: ['*']
  push:
    branches: ['master']

env:
  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

jobs:
  build:
    name: Build and Test
    if: "!contains(github.event.head_commit.message, 'ci skip')"
    strategy:
      matrix:
        os: [ubuntu-latest]
        scala: [2.12.10]
        java: [adopt@1.11]
    runs-on: ${{ matrix.os }}
    steps:
      - name: Checkout current branch (full)
        uses: actions/checkout@v2
        with:
          fetch-depth: 0

      - name: Setup Java and Scala
        uses: olafurpg/setup-scala@v5
        with:
          java-version: ${{ matrix.java }}

      - name: Cache ivy2
        uses: actions/cache@v1
        with:
          path: ~/.ivy2/cache
          key: ${{ runner.os }}-sbt-ivy-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}

      - name: Cache coursier (generic)
        uses: actions/cache@v1
        with:
          path: ~/.coursier/cache/v1
          key: ${{ runner.os }}-generic-sbt-coursier-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}

      - name: Cache coursier (linux)
        if: contains(runner.os, 'linux')
        uses: actions/cache@v1
        with:
          path: ~/.cache/coursier/v1
          key: ${{ runner.os }}-sbt-coursier-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}

      - name: Cache coursier (macOS)
        if: contains(runner.os, 'macos')
        uses: actions/cache@v1
        with:
          path: ~/Library/Caches/Coursier/v1
          key: ${{ runner.os }}-sbt-coursier-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}

      - name: Cache coursier (windows)
        if: contains(runner.os, 'windows')
        uses: actions/cache@v1
        with:
          path: ~/AppData/Local/Coursier/Cache/v1
          key: ${{ runner.os }}-sbt-coursier-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}

      - name: Cache sbt
        uses: actions/cache@v1
        with:
          path: ~/.sbt
          key: ${{ runner.os }}-sbt-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}

      - run: sbt ++${{ matrix.scala }} validate verify

      - name: Codecov
        uses: codecov/codecov-action@v1
        with:
          fail_ci_if_error: true
```

Comment (ptravers) [reply .github/workflows/ci.yml:23]: Ah cool! Will refactor!

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          key: ${{ runner.os }}-generic-sbt-coursier-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}
+
+      - name: Cache coursier (linux)
+        if: contains(runner.os, 'linux')
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/coursier/v1
+          key: ${{ runner.os }}-sbt-coursier-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}
+
+      - name: Cache coursier (macOS)
```
Comment (pauljamescleary) [inline .github/workflows/ci.yml:53]: you can delete the `macOS` and `windows` cache things here, my bad for putting them in (we don't build on mac or windows)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+name: Continuous Integration
+
+on:
+  pull_request:
+    branches: ['*']
+  push:
+    branches: ['master']
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (pauljamescleary) [inline .github/workflows/ci.yml:11]: this can go away as it is not used afaik

**Review by pauljamescleary — COMMENTED** [review]
A few small things that were mostly my fault and this is good :)

**Review by pauljamescleary — APPROVED** [review]
Nice @ptravers  :)

**Pull request merged**